### DISCUSSION TITLE
Manuals?

### DISCUSSION THREAD

**Discussion by DanRight** [epochpic/epoch#370] [Q&A] [answered]
Reading sources i find

    ! Weighting factors as Eqn 4.77 page 25 of manual
    ! Eqn 4.77 would be written as
    ! F(j-1) * gmx + F(j) * g0x + F(j+1) * gpx

Where is this manual? There is no mention of it in this git. 

Also is there a list and description of all major variables in the source code with their dimensions ? Particle items, their coordinates, momentum, Energy, charge, ID and field values. Would be very helpful to have one. "Users Manual for the EPOCH PIC codes" mostly devoted to setup input.deck does not have it, or i heavily miss something?

**Comment by PhysicsDan** [comment]
Here's the documentation website: https://epochpic.github.io/documentation.html

**Reply by DanRRRR** [reply]
Thanks!

**Comment by Status-Mirror** [comment, marked as answer]
Hi DanRight,

The documentation linked to by PhysicsDan contains the user-manual, which features lots of useful information about running EPOCH. However, the source-code comments describe content found in the developer's manual, which can be found here:

https://github.com/Warwick-Plasma/EPOCH_manuals/releases

Note that the page numbers in the most recent Developer's Manual are slightly different to those in the source code.

Hope this helps,
Stuart

**Comment by DanRight** [comment]
Thanks, StuartDan
    On Monday, September 26, 2022 at 07:11:36 AM PDT, Stuart Morris ***@***.***> wrote:  
 
 


Hi DanRight,

The documentation linked to by PhysicsDan contains the user-manual, which features lots of useful information about running EPOCH. However, the source-code comments describe content found in the developer's manual, which can be found here:

https://github.com/Warwick-Plasma/EPOCH_manuals/releases

Note that the page numbers in the most recent Developer's Manual are slightly different to those in the source code.

Hope this helps,
Stuart

**Comment by DanRRRR** [comment]
Hi Stuart,
Thanks for a lot of updates today. Great to see that this project is not in totally anemic state like it looked like last few months :)
I have some questions about physics of these updates
1) Looks like in ver.4.17 there was no collisional ionization, and now in ver 5 it is included? Am i wrong? And where crossections are taken from? From some formulas (like  Lotz) , atomic codes fitting expressions or tables?
2) Were photorecombination and three-body recombination included? 
3) Can the code run for any atom of periodic table ?
4) Ionization of low Z (close to neutral states of atoms and molecules) has its specific compared to ionization of ideal plasma Braginskii-Spitzer described. There exist strong attachment of electrons to atoms and molecules, and formation of negative ions etc. The charge there usually much lower than if just use classic plasma approaches. As a result the conductivity of such low Z plasma  depends on electron density and its value is way smaller than in classical case. In Braginskii case conductivity almost does not depend on density.  Was such low Z ionization physics of atoms and molecules included in last version, at least potentially? 
Best regardsDan
-----Original Message-----

**Reply by Status-Mirror** [reply]
Hey Dan,

It's interesting that there are 3 seemingly unrelated Dans in this discussion page!

Collisional ionisation has been in EPOCH for a while, but the version in 4.17, the current 4.18, and the developmental 5.0 branch is bugged. We are working on a major overhaul for 4.19 (which will automatically update the 5.0 development branch), and we are close to publishing our new method with all the details. Currently we are aiming to release 4.19 in December, with full documentation on the new collisional ionisation model.

To summarise the 4.19 model:

1.  We use a combination of the MBELL and RBEB models for collisional ionisation cross sections. MBELL is a fit to experimental data up to the M shell, so it's accurate, but it won't describe all elements. Above Z=18, we switch to RBEB.
2. Recombination is not included. Some groups are interested in adding this, but we are struggling to find good estimates for these rates for high energy particles in heated, solid-density targets (the TNSA regime, where collisional ionisation seems to be used the most). I know of PIC codes which include this, but they use a recombination rate which forces Saha-Boltzmann charge ratios, which I would imagine only works for targets in thermal equilibrium.
3. Currently we have binding energy data for elements up to Z=100. The Desclaux binding energy tables go higher, so this limit could be raised if there was a demand. Z=100 was chosen, as this is how high the bremsstrahlung tables go.
4. There is currently no special treatment for the low-Z near-neutral elements you describe. I know of a group in the University of York which has been developing a low-temperature version of EPOCH, with enhanced treatment of neutrals. This is a big change from the normal EPOCH code though, and there are currently no plans to merge the two.

Hope this helps,
Stuart

References:
MBELL: Haque, A. K. F., et al. "Electron impact ionization of M-shell atoms." Physica Scripta 74.3 (2006): 377.
RBEB: Kim, Yong-Ki, et al. "Extension of the binary-encounter-dipole model to relativistic incident electrons." Physical Review A 62.5 (2000): 052710.
BInding energies: Desclaux, J. P. "Relativistic Dirac-Fock expectation values for atoms with Z= 1 to Z= 120." Atomic data and nuclear data tables 12.4 (1973): 311-406.

**Comment by DanRight** [comment]
Hi Stuart,
May be some these Dans are me too. I remember i was confused with all these GitHubs and GitLabs, and though i forgot the usernames/passwords. 
Potentially we can think to add at least photorecombination for some specific cases based on detailed atomic code calculations.  To do that universally for all atomic elements and ionization stages is of course large work for years.
With 3-body recombination indeed things are not that clear. Good here is that in some cases the photorecombination might dominate in high Z  or low density plasmas.
As to low temperature plasma do you remember any names from this group Univ of York? 
Cheers,Dan


    On Monday, November 14, 2022 at 01:24:16 AM PST, Stuart Morris ***@***.***> wrote:  
 
 


Hey Dan,

It's interesting that there are 3 seemingly unrelated Dans in this discussion page!

Collisional ionisation has been in EPOCH for a while, but the version in 4.17, the current 4.18, and the developmental 5.0 branch is bugged. We are working on a major overhaul for 4.19 (which will automatically update the 5.0 development branch), and we are close to publishing our new method with all the details. Currently we are aiming to release 4.19 in December, with full documentation on the new collisional ionisation model.

To summarise the 4.19 model:
   
   - We use a combination of the MBELL and RBEB models for collisional ionisation cross sections. MBELL is a fit to experimental data up to the M shell, so it's accurate, but it won't describe all elements. Above Z=18, we switch to RBEB.
   - Recombination is not included. Some groups are interested in adding this, but we are struggling to find good estimates for these rates for high energy particles in heated, solid-density targets (the TNSA regime, where collisional ionisation seems to be used the most). I know of PIC codes which include this, but they use a recombination rate which forces Saha-Boltzmann charge ratios, which I would imagine only works for targets in thermal equilibrium.
   - Currently we have binding energy data for elements up to Z=100. The Desclaux binding energy tables go higher, so this limit could be raised if there was a demand. Z=100 was chosen, as this is how high the bremsstrahlung tables go.
   - There is currently no special treatment for the low-Z near-neutral elements you describe. I know of a group in the University of York which has been developing a low-temperature version of EPOCH, with enhanced treatment of neutrals. This is a big change from the normal EPOCH code though, and there are currently no plans to merge the two.

Hope this helps,
Stuart

References:
MBELL: Haque, A. K. F., et al. "Electron impact ionization of M-shell atoms." Physica Scripta 74.3 (2006): 377.
RBEB: Kim, Yong-Ki, et al. "Extension of the binary-encounter-dipole model to relativistic incident electrons." Physical Review A 62.5 (2000): 052710.
BInding energies: Desclaux, J. P. "Relativistic Dirac-Fock expectation values for atoms with Z= 1 to Z= 120." Atomic data and nuclear data tables 12.4 (1973): 311-406.

**Comment by Status-Mirror** [comment]
If there is demand for a specific recombination scenario, we may be able to provide a recombination algorithm in a separate developmental branch. A group in LMU are also interested in a recombination PIC algorithm. If we were to do this though, we'd need to be supplied with the recombination rates/cross-sections to use, and how they vary with incident particle and background plasma parameters.

The low temperature PIC code was done in a group led by Christopher Ridgers at York. He may be able to provide further details.
